### DISCUSSION TITLE
Disable signatures for overloads

### DISCUSSION THREAD

**Discussion by awni** [wjakob/nanobind#495] [Q&A]
Is there (or could there be) a way to disable signatures for overloads? It would be nice to have an option to set the signature parameter to a union type rather than have a list of overload signatures, but still use the overload machinery in nanobind for the bindings.

**Comment by wjakob** [comment]
You could remove them in stubs using the pattern file approach explained in the documentation.

**Reply by awni** [reply]
Cool thanks for the pointer, I will take a look. The main downside I foresee is we'd need to start building the docs from stubs rather than from source code directly.

Just throwing it out there (as I have no idea if it fits easily)... but it would be really nice if you could pass something like `nb::sig()` to switch off the signature for a given overload.

**Reply by wjakob** [reply]
I'd say this is out of scope. But should be easy to patch into a custom version of nanobind.
